netlink-sys/src/smol.rs

```diff
@@ -66,14 +66,14 @@ impl SmolSocket {
             trace!("poll_recv_from socket is ready for reading");
 
             match self.0.get_ref().recv_from(buf, 0) {
-                Ok(x) => {
-                    trace!("poll_recv_from {:?} bytes read", x);
-                    return Poll::Ready(Ok(x));
-                }
-                Err(_would_block) => {
+                Err(e) if e.kind() == io::ErrorKind::WouldBlock => {
```

Comment:
oops, thanks. Maybe we should still distinguish between `Err` and `Ok` below though, otherwise the trace message might be slightly confusing when there's an error.

Reply:
Afaict this was and is the same for both the tokio and smol code; I'd recommend to change this some other time. (Also this code is replaced in the `AsyncSocket` commit, but with similar trace logging.)